src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import debounce from 'lodash.debounce';
+import { Person } from '../../types/Person';
+import './Dropdown.scss';
+
+type Props = {
+  people: Person[];
+  selectPerson: Dispatch<SetStateAction<Person | null>>;
+  setIsMatched: Dispatch<SetStateAction<boolean>>;
+};
+
+export const Dropdown: React.FC<Props> = ({
+  people,
+  selectPerson,
+  setIsMatched,
+}) => {
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const [isFocused, setIsFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const filteredPeople = people.filter(({ name }) =>
+    name.toLowerCase().includes(appliedQuery.toLowerCase()),
+  );
+
+  const isMatched = filteredPeople.length > 0;
+
+  const handleSelectPerson = (name: string) => {
+    const selectedPerson = people.find(person => person.name === name);
+
+    if (selectedPerson) {
+      selectPerson(selectedPerson);
+      setIsFocused(false);
+    }
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setIsFocused(true);
+    setQuery(value);
+    applyQuery(value);
+    selectPerson(null);
+  };
+
+  useEffect(() => {
+    setIsMatched(isMatched);
+  }, [isMatched, setIsMatched]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsFocused(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
```

Comment:
`onBlur` doesn't work in your case?

Reply:
It works but with onBlur I can't do that.

https://github.com/mate-academy/react_autocomplete/assets/130569769/7d8c7eff-1d2c-4650-a4cd-5f5f79fa4e3d